src/framework/shortcuts/internal/platform/macos/macosshortcutsinstancemodel.mm

```diff
@@ -26,13 +26,65 @@
 #include <QApplication>
 #include <QKeyEvent>
 
-#include <private/qguiapplication_p.h>
-#include <qpa/qplatformintegration.h>
-
 #include "log.h"
 
 using namespace muse::shortcuts;
 
+static std::map<UInt32, QString> specialKeysMap = {
+    { kVK_F1, "F1" },
+    { kVK_F2, "F2" },
+    { kVK_F3, "F3" },
+    { kVK_F4, "F4" },
+    { kVK_F5, "F5" },
+    { kVK_F6, "F6" },
+    { kVK_F7, "F7" },
+    { kVK_F8, "F8" },
+    { kVK_F9, "F9" },
+    { kVK_F10, "F10" },
+    { kVK_F11, "F11" },
+    { kVK_F12, "F12" },
+    { kVK_F13, "F13" },
+    { kVK_F14, "F14" },
+    { kVK_F15, "F15" },
+    { kVK_F16, "F16" },
+    { kVK_F17, "F17" },
+    { kVK_F18, "F18" },
+    { kVK_F19, "F19" },
+    { kVK_Space, "Space" },
+    { kVK_Escape, "Esc" },
+    { kVK_Delete, "Delete" },
+    { kVK_ForwardDelete, "Backspace" },
+    { kVK_LeftArrow, "Left" },
+    { kVK_RightArrow, "Right" },
+    { kVK_UpArrow, "Up" },
+    { kVK_DownArrow, "Down" },
+    { kVK_Help, "" },
+    { kVK_PageUp, "PgUp" },
+    { kVK_PageDown, "PgDown" },
+    { kVK_Tab, "Tab" },
+    { kVK_Return, "Return" },
+    { kVK_Home, "Home" },
+    { kVK_End, "End" },
+    { kVK_ANSI_Keypad0, "0" },
+    { kVK_ANSI_Keypad1, "1" },
+    { kVK_ANSI_Keypad2, "2" },
+    { kVK_ANSI_Keypad3, "3" },
+    { kVK_ANSI_Keypad4, "4" },
+    { kVK_ANSI_Keypad5, "5" },
+    { kVK_ANSI_Keypad6, "6" },
+    { kVK_ANSI_Keypad7, "7" },
+    { kVK_ANSI_Keypad8, "8" },
+    { kVK_ANSI_Keypad9, "9" },
+    { kVK_ANSI_KeypadDecimal, "." },
```

Comment:
I believe this depends on the keyboard layout. In Dutch, it should produce `,`, and in most other languages `.`. Is that handled correctly in this implementation?

Reply:
Something weird with the comma in Dutch
On a test project using only Qt shortcuts when registering '.'/Num+./Num+ the shortcuts don't work, but on the other hand, the shortcuts work on 4.3.

Reply:
@cbjeukendrup @zacjansheski 
Let's merge this PR and log this issue separately

Reply:
Agree! As far as I'm concerned, it would be a P3/P4 issue, because the fraction of users who are on macOS _and_ have a numeric keypad must be very small nowadays, and the fraction of users who _use_ that numeric keypad must be even smaller.